api/subscriptions/subscriptions.go

```diff
@@ -387,15 +339,83 @@ func (s *Subscriptions) Close() {
 	s.wg.Wait()
 }
 
+func (s *Subscriptions) websocket(readerFunc func(http.ResponseWriter, *http.Request) (msgReader, error)) utils.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) error {
+		s.wg.Add(1)
+		defer s.wg.Done()
+
+		// Call the provided reader function
+		reader, err := readerFunc(w, req)
+		if err != nil {
+			return err
+		}
+
+		// Ensure cleanup when the connection closes
+		defer func() {
+			if closer, ok := reader.(interface{ Close() }); ok {
+				closer.Close()
+			}
+		}()
+
+		// Setup WebSocket connection
+		conn, closed, err := s.setupConn(w, req)
+		if err != nil {
+			logger.Debug("upgrade to websocket", "err", err)
+			return err
+		}
+		defer s.closeConn(conn, err)
+
+		// Stream messages
+		err = s.pipe(conn, reader, closed)
+		if err != nil {
+			logger.Debug("error in websocket pipe", "err", err)
+		}
+		return err
+	}
+}
+
+// handleGone is a handler for deprecated endpoints that returns HTTP 410 Gone.
+func handleGone(w http.ResponseWriter, _ *http.Request) error {
```

Comment:
Can be used in other deprecated endpoints? If so, maybe move it to utils folder?